Modules/Devices/test/Devices.Application.Tests/Tests/Extensions/IEnumerableExtensionsTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Backbone.Modules.Devices.Application.Extensions;
+using FluentAssertions;
+using Xunit;
+using static System.Double;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Extensions;
+public class IEnumerableExtensionsTests
+{
+    [Fact]
+    public void Null_Parameter_Throws_ArgumentNullException()
+    {
+        // Act
+        var acting = () => ((List<int>)null).Split(10); ;
+
+        // Assert
+        acting.Should().Throw<ArgumentNullException>();
+    }
+
+    [Fact]
+    public void Empty_List_Returns_Empty_List_Of_Lists()
+    {
+        // Arrange
+        var list = new List<int>().AsReadOnly();
+
+        // Act
+        var result = list.Split(10);
+
+        // Assert
+        result.Should().BeEmpty("The argument list was empty.");
```

Comment:
It's not necessary to pass a reason. The error messages of FluentAssertions are good enough.